manifests/init.pp

```diff
@@ -189,6 +189,7 @@
   String        $ip_address                         = '',
   String        $ipa_server_fqdn                    = $::fqdn,
   String        $ipa_master_fqdn                    = '',
+  Boolean       $ipa_force_join                     = false,
```

Comment:
This changes the logic on how this force-join feature was supposed to work. The idea behind using a fact was this: the *$::ipa_force_join* fact would be set at provisioning time (e.g. user-data). If the fact was set to *true*, Puppet would know that it needs to append *--force-join* to the ipa-client-install command-line. The fact would be removed after Puppet did its work and subsequent Puppet runs would not append *--force-join* to the command-line.

If you set the class parameter *$ipa_force_join = true* then all ipa-client-install incantations will have *--force-join* in them. Whether this has any practical implications I can't recall / don't know.

In any case this breaks some of our provisioning tooling. What I suggest is changing this to

```
Boolean $force_join = $::ipa_force_join
```
This means backwards compability is maintained and there's no way the parameter and fact names could accidentally collidate.